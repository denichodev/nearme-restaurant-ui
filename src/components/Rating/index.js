import React from 'react';
import { bool, number } from 'prop-types';
import cn from 'classnames';

import s from './styles.css';

const Rating = ({ point, big }) => {
  const style = {};

  if (point > 4) {
    style['backgroundColor'] = '#5BA829';
  } else if (point > 3 && point <= 4) {
    style['backgroundColor'] = '#9ACD32';
  } else if (point > 2 && point <= 3) {
    style['backgroundColor'] = '#CDD614';
  } else if (point > 0.1 && point <= 2) {
    style['backgroundColor'] = '#FFBA00';
  } else {
    style['backgroundColor'] = '#89959B';
  }

  const html =
    point > 0.1 ? (
      <div>
        <span className={s.point}>{point}</span>
        <span className={s.total}>/5</span>
      </div>
    ) : (
      <span className={s.new}>NEW</span>
    );

  return (
    <div className={cn(s.rating, big && s.big)} style={style}>
      {html}
    </div>
  );
};

Rating.propTypes = {
  point: number,
  big: bool,
};

export default Rating;
