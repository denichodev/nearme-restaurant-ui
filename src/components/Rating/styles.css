.rating {
  position: absolute;
  top: 0;
  right: 0;
  font-weight: 700;
  padding: 5px 8px;
  border-radius: 0 10px 0 10px;
}

.rating.big {
  font-size: 18px;
  padding: 10px 15px;
}

.point {
  font-size: 14px;
  color: #FFF;
}

.total {
  color: rgba(255, 255, 255, 0.7);
  font-size: 12px;
}

.new {
  font-size: 12px;
  color: rgba(255, 255, 255, 0.7);
}

.big .new {
  font-size: 20px;
}

.big .point {
  font-size: 20px;
}

.big .total {
  font-size: 14px;
}