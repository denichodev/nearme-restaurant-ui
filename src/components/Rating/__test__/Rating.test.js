import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Rating from '../index';

describe('<Rating />', () => {
  test('renders correctly', () => {
    const wrapper = shallow(
      <Rating point={4.5} />,
    );

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  test('has `.big` css class', () => {
    const wrapper = shallow(
      <Rating point={4.5} big />,
    );

    expect(wrapper.find('.big')).toHaveLength(1);
  });

  test('has correct background color, shows .new if rating is 0', () => {
    const wrapper = shallow(
      <Rating point={4.5} />,
    );

    expect(wrapper.prop('style')).toHaveProperty('backgroundColor', '#5BA829');
    expect(wrapper.find('.new')).toHaveLength(0);

    wrapper.setProps({ point: 3.6 })
    expect(wrapper.prop('style')).toHaveProperty('backgroundColor', '#9ACD32');
    expect(wrapper.find('.new')).toHaveLength(0);

    wrapper.setProps({ point: 2.5 })
    expect(wrapper.prop('style')).toHaveProperty('backgroundColor', '#CDD614');
    expect(wrapper.find('.new')).toHaveLength(0);

    wrapper.setProps({ point: 1.2 })
    expect(wrapper.prop('style')).toHaveProperty('backgroundColor', '#FFBA00');
    expect(wrapper.find('.new')).toHaveLength(0);

    wrapper.setProps({ point: 0 })
    expect(wrapper.find('.new')).toHaveLength(1);
  });
});