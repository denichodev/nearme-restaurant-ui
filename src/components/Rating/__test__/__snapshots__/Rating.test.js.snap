// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating /> renders correctly 1`] = `
<div
  className="rating"
  style={
    Object {
      "backgroundColor": "#5BA829",
    }
  }
>
  <div>
    <span
      className="point"
    >
      4.5
    </span>
    <span
      className="total"
    >
      /5
    </span>
  </div>
</div>
`;
